networkx/algorithms/approximation/traveling_salesman.py

```diff
@@ -265,13 +265,18 @@ def traveling_salesman_problem(G, weight="weight", nodes=None, cycle=True, metho
         List of nodes in `G` along a path with an approximation of the minimal
         path through `nodes`.
 
-
     Raises
     ------
     NetworkXError
         If `G` is a directed graph it has to be strongly connected or the
         complete version cannot be generated.
 
+    Notes
+    -----
+    If the input graph `G` contains edges whose weights do not satisfy the
+    triangle inequality, then the returned path may contain repeating nodes
+    (other than the starting node).
```

Comment:
We should also include the case where the graph is not a complete graph (where the triangle inequality definitely doesn't apply -- non-existing edges are length `inf` so definitely violate the inequality).  Perhaps add a phrase like: `e.g. when a graph is not a complete graph.`

And perhaps this should appear at the end of the paragraph at line 222 instead of in a separate `Notes` section.